### PR TITLE
[14.0][IMP] web_field_required_invisible_manager: compatibility with tier v…

### PR DESCRIPTION
**Pull request by ilyasProgrammer**
…alidation

moved code from base model to ir.ui.view + minor refactoring.

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

File: web_field_required_invisible_manager/models/models.py [hunk 1]
```diff
@@ -27,34 +27,32 @@ class IrModel(models.Model):
...
-            return arch
-        # TODO speed up somehow
+    def postprocess_and_fields(self, node, model=None, validate=False):
+        arch, new_fields = super().postprocess_and_fields(node, model, validate)
+        if self.type not in ["form", "tree"] and node.tag not in ["form", "tree"]:
+            return arch, new_fields
         restrictions = self.env["custom.field.restriction"].search(
             [
-                ("model_name", "=", self._name),
+                ("model_name", "=", self.model or model),
```
Comment (HekkiMelody) [inline web_field_required_invisible_manager/models/models.py]: **issue**: from base Odoo, it looks like `model` should have priority over `self.model`

https://github.com/odoo/odoo/blob/14.0/odoo/addons/base/models/ir_ui_view.py#L857

File: web_field_required_invisible_manager/models/models.py [hunk 2]
```diff
@@ -27,34 +27,32 @@ class IrModel(models.Model):
...
                 ("group_ids", "in", self.env.user.groups_id.ids),
             ]
         )
+        view_type = self.type or node.tag
```
Comment (HekkiMelody) [inline web_field_required_invisible_manager/models/models.py]: **question**: I feel like here, too, `node.tag` should have priority over `self.type` (e.g.: a tree view embedded in a form view) but I'm not sure. What do you think?

**Review by HekkiMelody — CHANGES_REQUESTED** [review]
Mostly ok, one comment and one question I'm not sure about.

Since you're already doing some cleanup and refactoring, could you please put each model in its own file, named after the model itself?

Thank you!

File: web_field_required_invisible_manager/models/models.py (hunk 2 above)
Comment (ilyasProgrammer) [reply web_field_required_invisible_manager/models/models.py]: i see no difference. But ok.

**Review by HekkiMelody — APPROVED** [review]

**Pull request merged**